core/blocksigner/enclave.go

```diff
@@ -0,0 +1,67 @@
+package blocksigner
+
+import (
+	"context"
+	"fmt"
+
+	"chain/core/rpc"
+	"chain/crypto/ed25519"
+	"chain/encoding/json"
+	"chain/log"
+	"chain/protocol/bc/legacy"
+)
+
+// EnclaveClient implements the Signer interface by calling
+// Chain Enclave to sign blocks.
+type EnclaveClient struct {
+	// URLs is called on every Sign call to retrieve the URLs
+	// and access tokens for Chain Enclave.
+	URLs       func() [][]string
+	BaseClient rpc.Client
+}
+
+type signRequestBody struct {
+	Block *legacy.BlockHeader `json:"block"`
+	Pub   json.HexBytes       `json:"pubkey"`
+}
+
+func (ec EnclaveClient) Sign(ctx context.Context, pk ed25519.PublicKey, bh *legacy.BlockHeader) ([]byte, error) {
+	body := signRequestBody{Block: bh, Pub: json.HexBytes(pk[:])}
+
+	// grab the latest set of hsms from the configuration
+	hsmURLs := ec.URLs()
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ch := make(chan result, len(hsmURLs))
+	for _, tup := range hsmURLs {
+		// make a copy of the base rpc client on the stack so we
+		// can modify it with the URL and access token
+		client := ec.BaseClient
+		client.BaseURL = tup[0]
+		client.AccessToken = tup[1]
+
+		go func(client rpc.Client) {
+			var signature []byte
+			err := client.Call(ctx, "/sign-block", body, &signature)
+			ch <- result{signature: signature, err: err}
+		}(client)
```

Comment:
Nit: since client is declared inside the loop body, it doesn't need to be a param here.